gocourse02/main.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import "fmt"
+
+// Struct that represents an Animal
+type Animal struct {
+	name      string // Name of the animal
+	species   string // Species of the animal
+	inTheCage bool   // Flag indicating whether the animal is in a cage
+}
+
+// Struct that represents a Zoo, containing multiple animals
+type Zoo struct {
+	Animals []Animal // Slice of animals in the zoo
+}
+
+// Struct that represents a Cage, which can hold an animal
+type Cage struct {
+	name   string // Name of the cage
+	animal string // Species of the animal currently in the cage
```

Comment:
А ми можемо тут замість назви тварини зберігати поінтер на тварину?

```suggestion
	animal *Animal // the animal currently in the cage
```

Поінтер буде `nil`, якщо клітка не зайнята.

Reply:
Done